mono/metadata/appdomain.c

```diff
@@ -2198,13 +2198,19 @@ mono_domain_assembly_search (MonoAssemblyName *aname,
 	GSList *tmp;
 	MonoAssembly *ass;
 	gboolean refonly = GPOINTER_TO_UINT (user_data);
+	const gboolean strong_name = aname->public_key_token[0] != 0;
```

Comment:
I wasn't sure about this check and had to look around. We should extra this into a `mono_assembly_name_has_pubkey` and cleanup our code.